modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -210,11 +255,63 @@ def _read(cls, filepath_or_buffer, **kwargs):
             elif isinstance(parse_dates, dict):
                 for new_col_name, group in parse_dates.items():
                     column_names = column_names.drop(group).insert(0, new_col_name)
+
+        return column_names, column_widths
+
+    @classmethod
+    def _get_new_qc(
+        cls,
+        partition_ids,
+        index_ids,
+        dtypes_ids,
+        index_col_md,
+        index_name,
+        column_widths,
+        column_names,
+        **kwargs,
+    ):
```

Comment:
Would it be possible to add type hints here or does type hinting not play nice with `new_query_compiler`?

Reply:
It can be done for `PandasQueryCompiler`, but we also have to add experimental query compiler, that will cause `ImportError` in environments where experimental dependencies are not installed.